obspy/core/event/event.py

```diff
@@ -292,6 +292,11 @@ def __deepcopy__(self, memodict=None):
         result.resource_id.bind_resource_ids()  # bind all resource_ids
         return result
 
+    @rlock
+    def __setstate__(self, state_dict):
+        super(Event, self).__setstate__(state_dict)
+        ResourceIdentifier.bind_resource_ids()
```

Comment:
Just writing down my thoughts: There are now two separate thread locks for the Event objects but I think this if fine as `__setstate__()` and `__deepcopy__()` cannot really be called at the same time so there are no race conditions.

Reply:
Should `bind_resource_ids()` also be called for all the other event objects like `Origin`, `Magnitude`, ...?

Reply:
Tbh I am not really sure if we need an rlock on `__setstate__` or not. When I am feeling really ambitious I should setup a nasty multi-threaded test case to make sure it all still works with many threads.